backend/functions/src/routes/students/index.ts

```diff
@@ -0,0 +1,106 @@
+import express from 'express';
+import { firebaseApp } from '../../index';
+
+const students = express();
+interface student {
+  firstName: string;
+  lastName: string;
+  gradYear: number;
+  email: string;
+  netid: string;
+  misc?: object;
+}
+
+function isStudent(data: object): data is student {
+  return 'firstName' in data && 'lastName' in data && 'gradYear' in data;
+}
+
+/**
+ * Creates a new student in the `registrants` collection using x-www-form-urlencoded data.
+ *
+ * Upon success returns the document ID with HTTP 202
+ * Upon fail returns HTTP 400
+ */
+students.post('/createStudent', async (req, res) => {
+  let studentData = req.body;
+
+  if (!isStudent(studentData)) {
+    res.status(400).send({
+      error: 'Malformed student registration request.'
+    });
+    return;
+  }
+
+  await firebaseApp
+    .firestore()
+    .collection('students')
+    .doc(studentData.email)
+    .set(studentData);
+
+  res.sendStatus(200);
+});
+
+/**
+ * Updates a student with new parameters. Does not support updating arrays yet.
+ */
+students.post('/updateStudent', async (req, res) => {
+  let studentData = req.body;
+
+  // Check that email field exists on req.body
+  if (!('email' in studentData)) {
+    res.status(400).send({
+      error: 'Student email not specified.'
+    });
+  }
+
+  // TODO: handle food allergies with firestore arrayUnion()
+
+  await firebaseApp
+    .firestore()
+    .collection('students')
+    .doc(studentData.email)
+    .update(studentData);
+
+  res.sendStatus(200);
+});
+
+/**
+ * Retrieves student data given a supplied email.
+ */
+students.get('/getStudent', async (req, res) => {
+  let {email} = req.query;
```

Comment:
Similary to above, I think this should be a `/:email` endpoint instead, where the email in the path tells you which student to retrieve.